src/node_api.cc

```diff
@@ -706,6 +706,40 @@ bool FindWrapper(v8::Local<v8::Object> obj,
   return true;
 }
 
+static void DeleteEnv(const v8::WeakCallbackInfo<napi_env__>& data) {
+  delete data.GetParameter();
+}
+
+napi_env get_env(v8::Local<v8::Context> context) {
+  napi_env result;
+
+  auto isolate = context->GetIsolate();
+  auto global = context->Global();
+
+  auto maybe_key = v8::String::NewFromUtf8(isolate, "N-API Environment",
+      v8::NewStringType::kInternalized);
+  CHECK(!maybe_key.IsEmpty());
+  auto key = v8::Private::ForApi(isolate, maybe_key.ToLocalChecked());
```

Comment:
If module loading performance isn't a concern, I wouldn't worry about it. FWIW it's also not necessary to use a mutex as `ForApi` is idempotent.